src/pages/profile/VerifyAirbnb.js

```diff
@@ -27,56 +33,76 @@ class VerifyAirbnb extends Component {
+        { this.state.mode === 'input-airbnb-profile' ? this.renderInputAirbnbProfile() : this.renderShowGeneratedCode() }
       </Modal>
     )
   }
 
-  renderErrorMessage(error) {
-    return (
-      <div className="form-group">
-        <div className="explanation">{error}</div>
-        <div className="button-container">
-            <button className="btn btn-clear" data-modal="airbnb"
-              onClick={e => {
-                e.preventDefault()
-                // reset state
-                this.setState({ mode: 'input-airbnb-profile', airbnbProfile: '', confirmationCode: '', error: '' })
-                this.props.handleToggle(e)
-              }}>
-              <FormattedMessage
-                id={ 'VerifyAirbnb.ok' }
-                defaultMessage={ 'Ok' }
-              />
-            </button>
-          </div>
-      </div>
-    )
-  }
-
-  renderNormalFlow(){
-    return(
-      this.state.mode === 'input-airbnb-profile' ? this.renderInputAirbnbProfile() : this.renderShowGeneratedCode()
-    )
-  }
-
   renderInputAirbnbProfile(){
+    const airbnbUserIdError = this.state.formErrors.airbnbUserId
     return(
        <form
-        onSubmit={async e => {
-          e.preventDefault()
+        onSubmit={async event => {
+          await this.catchPossibleErrors(async event => {
+            event.preventDefault()
+            this.clearErrors()
 
-          this.setState({ mode: 'show-generated-code', confirmationCode: '' })
+            const data = await origin.attestations.airbnbGenerateCode({
+              wallet: this.props.wallet,
+              airbnbUserId: this.getUserIdFromAirbnbProfile(this.state.airbnbProfile)
+            })
 
-          origin.attestations.airbnbGenerateCode({
-            wallet: this.props.wallet,
-            airbnbUserId: this.getUserIdFromAirbnbProfile(this.state.airbnbProfile)
-          }).then(data => {
-            this.setState({ confirmationCode: data.code })
-          })
+            this.setState({mode: 'show-generated-code', confirmationCode: data.code})
+          }, event)
         }}>
 
-        {this.renderProfileInputForm()}
+        <div className="form-group">
+          <label htmlFor="airbnbProfile">
+            { <FormattedMessage
+              id={ 'VerifyAirbnb.enterAirbnbProfileUrl' }
+              defaultMessage={ 'Enter Airbnb profile Url below' }
+            /> }
+          </label>
+          <div 
+            className={`form-control-wrap ${airbnbUserIdError ? 'error' : ''}`} 
+            style={{'maxWidth': `400px`}}
+          >
+            <input
+              type="url"
+              className="form-control"
+              id="airbnbProfile"
+              name="airbnbProfile"
+              // Making wider input, so that the whole profile placeholder can be viewed without trimming
+              style={{maxWidth: `400px`}}
```

Comment:
I see that this was already in the code, but could this be moved to a CSS class like `wide-input` or similar?